src/pages/Main/SearchResult.js

```diff
@@ -0,0 +1,140 @@
+import React, { Component } from 'react';
+import Modal from '../../components/Modal';
+import AddToCart from '../Products/Components/AddToCart';
+import Lists from '../Products/Components/Lists';
+import { PRODUCT_API } from '../../config';
+import Nav from '../../components/Nav';
+
+import './SearchResult.scss';
+
+class Products extends Component {
+  state = {
+    selectedOption: '',
+    productLists: [],
+    visibleCards: 8,
+    isModalAlertOpen: false,
+    isModalCartOpen: false,
+    selectedProduct: [],
+  };
+
+  componentDidMount() {
+    const keyword = this.props.location.search;
+    this.setState({ selectedOption: keyword.slice(9, keyword.length) });
+    fetch(`${PRODUCT_API}/products${keyword}`)
+      .then(res => res.json())
+      .then(searchdata =>
+        this.setState({
+          productLists: searchdata.product_info,
+        })
+      );
+  }
+
+  handleLoadMoreBtn = () => {
+    const { visibleCards } = this.state;
+    this.setState({
+      visibleCards: visibleCards + 8,
+    });
+  };
+
+  toggleModalAlert = () => {
+    const { isModalAlertOpen } = this.state;
+    this.setState({
+      isModalAlertOpen: !isModalAlertOpen,
+    });
+  };
+
+  toggleModalCart = id => {
+    const { isModalCartOpen } = this.state;
+    this.setState({
+      isModalCartOpen: !isModalCartOpen,
+    });
+
+    const url = '/data/selectedProduct.json'; //전달받은 id로 데이터 받아오기
+    fetch(url)
+      .then(res => res.json())
+      .then(data => data.product[0])
+      .then(selectedProduct =>
+        this.setState({
+          selectedProduct,
+        })
+      );

```

Comment:
```suggestion
      .then(data => this.setState({
        selectedProduct: data.product[0]
      }));
```
불필요한 then을 한번 더 늘릴 필요 없을 것 같습니다.